Sources/PaylikeEngine/ViewModel/Engine/PaylikeEngineStateHandling.swift

```diff
@@ -0,0 +1,104 @@
+import PaylikeClient
+import PaylikeRequest
+
+extension PaylikeEngine {
+    
+    /**
+     * Resetting engine field to default
+     */
+    public func resetEngine() {
+        loggingFn(Loggingformat(t: "Resetting engine", state: self.state))
+        
+        state = EngineState.WAITING_FOR_INPUT
+        error = nil
+        repository = EngineReposity()
+        webViewModel?.dropWebView()
+        
+        loggingFn(Loggingformat(t: "Resetted engine", state: self.state))
+
+        objectWillChange.send()
+    }
+    
+    /**
+     * Resetting engine field to default on MainActor
+     */
+    public func resetEngine() async {
+        await MainActor.run {
+            resetEngine()
+        }
+    }
+    
+    func saveState(newState: EngineState) {
+        state = newState
+        objectWillChange.send()
+    }
+    func saveState(newState: EngineState) async {
```

Comment:
Why do we have 2 separate save functions for each `save` function?

Reply:
When we change published fields, they should be on the main thread. The two kinds of save functions were the basic and the main threaded. But as I started to think about this solution, I thought of reorganizing this behaviour.

Calling something in the right thread is the caller's responsibility, right? So the engine only should provide basic saving functions, and then the caller has to call them at the right thread and pace, right?